### PR TITLE
Add support for trace filters UI in the dashboard

### LINKED ISSUE #5100: Custom filtering by attributes in Traces
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

_No response_

### Describe the solution you'd like

Logs can be filtered by record attributes. I would like to filter also Traces and Metrics records like logs.
Thank You

### Additional context

Here is my context explained:
https://github.com/dotnet/aspire/issues/4103#issuecomment-2242326164

### PR DESCRIPTION
**Pull request by JamesNK**
Structured logs support advanced filtering by tags. This PR adds similar functionality to distributed tracing

* Add filters UI to traces page
* Reuses FilterDialog. The dialog has been updated to be reused from both screens
* Adds tests to telemetry repository for filtering by tags

Fixes https://github.com/dotnet/aspire/issues/5100

Demo:
trace-filters

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5751)

### CODE REVIEW

**Comment by davidfowl** [comment]
@radical Why don't we get a dump on test timeouts:

```
 Aborting test run: test run timeout of 1200000 milliseconds exceeded.
  Aborting test run: test run timeout of 1200000 milliseconds exceeded.
  Results File: /datadisks/disk1/work/B7A10A6D/w/AECC0994/uploads/logs/TestResults.trx
  
  Test Run Aborted.
  Total tests: Unknown
       Passed: 20
   Total time: 20.0316 Minutes
/datadisks/disk1/work/B7A10A6D/w/AECC0994/e/tests/Aspire.Playground.Tests/bin/Debug/net8.0/Aspire.Playground.Tests.dll : error TESTRUNABORT: Test Run Aborted. [/datadisks/disk1/work/B7A10A6D/w/AECC0994/e/tests/Aspire.Playground.Tests/Aspire.Playground.Tests.csproj]
```

**Comment by davidfowl** [comment]
Instead of using xunit test timeouts maybe we should use dotnet test timeouts.

**Comment by radical** [comment]
> @radical Why don't we get a dump on test timeouts:
> 
> ```
>  Aborting test run: test run timeout of 1200000 milliseconds exceeded.
>   Aborting test run: test run timeout of 1200000 milliseconds exceeded.
>   Results File: /datadisks/disk1/work/B7A10A6D/w/AECC0994/uploads/logs/TestResults.trx
>   
>   Test Run Aborted.
>   Total tests: Unknown
>        Passed: 20
>    Total time: 20.0316 Minutes
> /datadisks/disk1/work/B7A10A6D/w/AECC0994/e/tests/Aspire.Playground.Tests/bin/Debug/net8.0/Aspire.Playground.Tests.dll : error TESTRUNABORT: Test Run Aborted. [/datadisks/disk1/work/B7A10A6D/w/AECC0994/e/tests/Aspire.Playground.Tests/Aspire.Playground.Tests.csproj]
> ```

We were getting it earlier. We pass `--blame-crash --blame-crash-dump-type full` to the test command. Maybe something broke in the update to the 9.0 tooling.

**Comment by radical** [comment]
Also, I think the playground tests are now running close to 20mins, so might be a good time to split that into multiple helix jobs.

File: src/Aspire.Dashboard/Components/Pages/Traces.razor.cs [hunk 1]
```diff
@@ -267,21 +272,88 @@ public void UpdateViewModelFromQuery(TracesPageViewModel viewModel)
...
+            {
+                TracesViewModel.ClearFilters();
+                foreach (var filter in filters)
+                {
+                    TracesViewModel.AddFilter(filter);
+                }
+            }
+        }
+
+        _ = Task.Run(async () =>
```
Comment (adamint) [inline src/Aspire.Dashboard/Components/Pages/Traces.razor.cs:290]: should we make the parent method async?

**Review by adamint — APPROVED** [review]
I like this a lot and couldn't find any issues in local testing. sorry for delayed review

File: src/Aspire.Dashboard/Components/Pages/Traces.razor.cs (hunk 1 above)
Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Pages/Traces.razor.cs:290]: I copied this from some other UI that did the same thing. I'll come back and look at improving both places later.

**Pull request merged**